### ISSUE TITLE
Improve the message output when a second administrator edits the already connected GTM module

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#4956]
## Bug Description

When an administrator, who doesn't have access to a Site Kit placed GTM property, tries to edit the module they're presented with the following error:
`Error: Not found or permission denied.`

This is due to another administrator having already connected the GTM module, inserting a GTM container snippet that other admins may not have access to. 

Improve this error message to provide further insights into this permissions issue. Something along the lines of what is shown (at present) for the same issue in Analytics may be suitable.

_Error: We've detected there's already an existing Google Tag Manager snippet on your site (GTM-*****), but your account doesn't seem to have access to this container. You can either remove the existing container snippet, or request access to this container from your team._

https://user-images.githubusercontent.com/41326532/159338105-ff551db3-32f3-4a01-a2c8-7385806b10e2.mp4

[Full recording of experience](https://recordit.co/weEaaxle4z)

## Steps to reproduce

1. Connect SK and the GTM module from one administrator account (admin1)
2. Logout and login from another administrator account (admin2)
3. Setup the plugin with admin2
4. Try to edit the Google Tag Manager module from admin2. There is no indication of the already connected property, instead it allows admin2 to connect to their own preferred property. This can result in Google Analytics or other data no longer connected if admin1 had this setup from GTM. 

## Additional Context

 - SK 1.70.0

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
@jamesozzie this will probably no longer be relevant with the work happening in the Existing Tag Simplification epic as all permission-related checks are going away. It's odd that this doesn't work similar to Analytics, I don't think this was the case before though.

**Comment by jamesozzie** [comment]
Just adding here that this notice can also appear if the connected GTM account was deleted at Google Tag Manager level. 

It does not appear if the container alone has been deleted, only if the account which is parent to that container was removed. If the container alone is deleted a user will see a blank  "Container" selected in the respective dropdown.

**Comment by aaemnnosttv** [comment]
This issue should no longer be relevant after #5495. Closing in favor of the changes already outlined there.
